### PR TITLE
LPS-193678 - Token renewal for salesforce authentication

### PR DESCRIPTION
**Pull request by G-Sa**
Related to [LPS-193678](https://liferay.atlassian.net/browse/LPS-193678)

### CODE REVIEW

File: modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/web/cache/SalesforceAccessTokenWebCacheItem.java [hunk 1]
```diff
@@ -29,13 +29,31 @@ public class SalesforceAccessTokenWebCacheItem implements WebCacheItem {
 	public static JSONObject get(
 		SalesforceConfiguration salesforceConfiguration) {
 
-		return (JSONObject)WebCachePoolUtil.get(
-			StringBundler.concat(
-				SalesforceAccessTokenWebCacheItem.class.getName(),
-				StringPool.POUND, salesforceConfiguration.consumerKey(),
-				StringPool.POUND, salesforceConfiguration.consumerSecret(),
-				StringPool.POUND, salesforceConfiguration.username()),
+		String salesforceConfigurationKey = StringBundler.concat(
```
Comment (carolmariaabb) [inline modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/web/cache/SalesforceAccessTokenWebCacheItem.java:32]: @G-Sa please rename it to `key`

File: modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/web/cache/SalesforceAccessTokenWebCacheItem.java [hunk 2]
```diff
@@ -29,13 +29,31 @@ public class SalesforceAccessTokenWebCacheItem implements WebCacheItem {
...
+			SalesforceAccessTokenWebCacheItem.class.getName(), StringPool.POUND,
+			salesforceConfiguration.consumerKey(), StringPool.POUND,
+			salesforceConfiguration.consumerSecret(), StringPool.POUND,
+			salesforceConfiguration.username());
+
+		JSONObject jSONObject = (JSONObject)WebCachePoolUtil.get(
```
Comment (carolmariaabb) [inline modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/web/cache/SalesforceAccessTokenWebCacheItem.java:38]: @G-Sa please rename it to `jsonObject`

**Review by carolmariaabb — CHANGES_REQUESTED** [review]
@G-Sa I requested some minor changes, as you can see below, but I believe we should ask @guilhermedcamacho to review this PR. 

@guilhermedcamacho could you please review this PR? Thanks!

File: modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/web/cache/SalesforceAccessTokenWebCacheItem.java [hunk 3]
```diff
@@ -29,13 +29,31 @@ public class SalesforceAccessTokenWebCacheItem implements WebCacheItem {
...
+
+		JSONObject jsonObject = (JSONObject)WebCachePoolUtil.get(
+			key,
 			new SalesforceAccessTokenWebCacheItem(salesforceConfiguration));
+
+		if (jsonObject == null) {
+			return null;
+		}
+
+		if ((jsonObject.getLong("issued_at") + _REFRESH_TIME) <
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/web/cache/SalesforceAccessTokenWebCacheItem.java:46]: Hey @G-Sa , were you able to reproduce the error and then test your fix?

If I understood correctly, your fix remove the item from cache when the issued timestamp + refresh time is before the current timestamp, so this way a new token is generated. However the [ehcache](https://www.ehcache.org/documentation/2.8/configuration/data-life.html) already do something similar, so if the timeToLive expires, it's will be removed from cache, this way a new token will be generated when method `WebCachePoolUtil#get` is invoked. You can change the _REFRESH_TIME to only 1 minute and test this behavior.
Let me know if I understood something wrong.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/web/cache/SalesforceAccessTokenWebCacheItem.java (hunk 3 above)
Comment (G-Sa) [reply modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/web/cache/SalesforceAccessTokenWebCacheItem.java:46]: Hi @guilhermedcamacho, i was able to reproduce the error, and even checked if the `WebCachePoolUtil#get` would remove the previous token after it's `timeToLive` expired, but i was still getting the same token sometimes (Older than the `timeToLive` defined).
I couldn't find the code responsible for removing the token from cache, that's why i went with this approach.

Comment (guilhermedcamacho) [reply modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/web/cache/SalesforceAccessTokenWebCacheItem.java:46]: Hey @G-Sa , as we discussed today, the right solution is to implement a retry logic.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by guilhermedcamacho** [comment]
See #2816

**Pull request closed without merging**